tap_s3_csv/__init__.py

```diff
@@ -72,14 +75,25 @@ def main():
     args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
     config = args.config
 
+    external_source = False
+
+    if 'external_id' in config:
+        args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS_EXTERNAL_SOURCE)
+        config = args.config
+        external_source = True
+
     config['tables'] = validate_table_config(config)
 
-    # Check that boto can access S3
-    try:
-        for page in s3.list_files_in_bucket(config['bucket']):
-            break
-    except err:
-        LOGGER.error(err)
+    # If external_id is provided, we are trying to access files in another AWS account, and need to assume the role
+    if external_source:
+        s3.setup_aws_client(config)
+    # Otherwise, confirm that we can access the bucket in our own AWS account
+    else:
+        try:
+            for page in s3.list_files_in_bucket(config['bucket']):
+                break
+        except BaseException as err:
+            LOGGER.error(err)
```

Comment:
For WP-7630

Reply:
this is sorta weird, i know it was there before.  is it just to check permissions?

Reply:
Yes, just confirms that you can access the bucket.